lib/game_teams_manager.rb

```diff
@@ -59,4 +59,117 @@ def add_wins_losses(gt_results, coach_record_start)
     end
     coach_record_start
   end
+
+#-------------TeamStats
+  def best_season(team_id)
+    best_season = win_percentage_by_season(team_id).max_by do |season, wins_percent|
+        wins_percent
+      end
+      best_year = best_season[0].to_i
+      "#{best_year}201#{best_year.digits[0] + 1}"
+    end
+
+  def worst_season(team_id)
+    worst_season = win_percentage_by_season(team_id).min_by do |season, wins_percent|
+        wins_percent
+      end
+      worst_year = worst_season[0].to_i
+      "#{worst_year}201#{worst_year.digits[0] + 1}"
+  end
+
+  def average_win_percentage(team_id)
+    (result_totals_by_team(team_id)[:wins].to_f / result_totals_by_team(team_id)[:total].to_f).round(2)
+  end
+
+  def most_goals_scored(team_id)
+    game_info_by_team(team_id).max_by do |game|
+      (game.goals).to_i
+    end.goals.to_i
+  end
+
+  def fewest_goals_scored(team_id)
+    game_info_by_team(team_id).min_by do |game|
+      (game.goals).to_i
+    end.goals.to_i
+  end
+
+  def favorite_opponent(team_id)
+    opponent = find_opponent_win_percentage(team_id).min_by do |team_id, percentage|
+      percentage
+    end.first
+    find_team_name(opponent)
+  end
+
+  def rival(team_id)
+    opponent = find_opponent_win_percentage(team_id).max_by do |team_id, percentage|
+      percentage
+    end.first
+    find_team_name(opponent)
+  end
+
+#-------------TeamStatsHelpers
+  def game_info_by_team(team_id)
+    @game_teams.select do |game_team|
+      game_team.team_id == team_id
+    end
+  end
+
+  def team_games_by_season(team_id)
+    team_games_by_season = {}
+    game_info_by_team(team_id).each do |game|
+      (team_games_by_season[game.game_id.to_s[0..3]] ||= []) << game
+    end
+    team_games_by_season
+  end
+
+  def win_percentage_by_season(team_id)
```

Comment:
This method is over 10 lines - would love to refactor this to be more succinct!

Reply:
We can talk bout it during refactoring, but I think it's fine for now!